operator/cluster/clusterlogic.go

```diff
@@ -199,9 +207,10 @@ func AddCluster(clientset *kubernetes.Clientset, client *rest.RESTClient, cl *cr
 		log.Info("primary Deployment " + cl.Spec.Name + " in namespace " + namespace + " already existed so not creating it ")
 	}
 
-	cl.Spec.UserLabels[config.LABEL_CURRENT_PRIMARY] = cl.Spec.Name
-
-	err = util.PatchClusterCRD(client, cl.Spec.UserLabels, cl, namespace)
+	//patch in the correct current primary value to the CRD spec, as well as
+	//any updated user labels. This will handle both new and updated clusters.
+	//Note: in previous operator versions, this was stored in a user label
+	err = util.PatchClusterCRD(client, cl.Spec.UserLabels, cl, primaryName, namespace)
 	if err != nil {
```

Comment:
I'm not entirely convinced we need to add `primaryName` as part of the `PatchClusterCRD` arguments. Would the primaryName be set in the annotation already?

Also (and while this is not on you), I'd recommend doing the:

```go
if err := util.PatchClusterCRD(...); err != nil {
```
since this is being edited anyway :smile:

Reply:
Since the PatchClusterCRD function is using a JSON Merge Patch, my understanding is that it works similar to a diff file, so that if the CRD being passed in already contains the correct value, the patch will have no effect (if I'm in some way mistaken on that, please let me know). There may be some value in revamping the CRD patching in general, but that may be out of scope for this PR.

As for the second change, happy to put that in place.